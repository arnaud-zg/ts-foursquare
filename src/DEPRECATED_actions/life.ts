import { ActionType, createAction, createAsyncAction } from 'typesafe-actions'
import { NRequest, NResolve } from '../../types'
import { adaptGetAccessToken } from '../adapter'

export const setOAuth2 = createAction('SET_OAUTH2')<NResolve.IOAuth2>()

export const getAccessTokenAsync = createAsyncAction(
  'GET_ACCESS_TOKEN_REQUEST',
  'GET_ACCESS_TOKEN_SUCCESS',
  'GET_ACCESS_TOKEN_FAILURE',
  'GET_ACCESS_TOKEN_CANCEL'
)<
  NRequest.IOAuth2AccessTokenPayload,
  ReturnType<typeof adaptGetAccessToken>,
  Error,
  string
>()

export const putCredentials = createAction('PUT_CREDENTIALS')<
  NResolve.ICredentialsPayload
>()

export const lifeActions = {
  setOAuth2,
  getAccessTokenAsync,
  putCredentials,
}

export type TLifeAction = ActionType<typeof lifeActions>
