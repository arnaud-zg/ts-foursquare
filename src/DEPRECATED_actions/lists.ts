import { ActionType, createAsyncAction } from 'typesafe-actions'
import { NRequest } from '../../types'
import { adaptGetLists } from '../adapter'

export const getListsAsync = createAsyncAction(
  'GET_LISTS_REQUEST',
  'GET_LISTS_SUCCESS',
  'GET_LISTS_FAILURE',
  'GET_LISTS_CANCEL'
)<NRequest.IListsPayload, ReturnType<typeof adaptGetLists>, Error, string>()

export const listsActions = {
  getListsAsync,
}

export type TListsAction = ActionType<typeof listsActions>
