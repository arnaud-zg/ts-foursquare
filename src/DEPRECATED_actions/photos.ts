import { ActionType, createAsyncAction } from 'typesafe-actions'
import { NRequest } from '../../types'
import { adaptGetPhotosDetails } from '../adapter'

export const getPhotosDetailsAsync = createAsyncAction(
  'GET_PHOTOS_DETAILS_REQUEST',
  'GET_PHOTOS_DETAILS_SUCCESS',
  'GET_PHOTOS_DETAILS_FAILURE',
  'GET_PHOTOS_DETAILS_CANCEL'
)<
  NRequest.IPhotosDetailsPayload,
  ReturnType<typeof adaptGetPhotosDetails>,
  Error,
  string
>()

export const photosActions = {
  getPhotosDetailsAsync,
}

export type TPhotosAction = ActionType<typeof photosActions>
