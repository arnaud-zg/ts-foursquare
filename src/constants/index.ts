export const PROJECT_NAME = 'ts-foursquare'
