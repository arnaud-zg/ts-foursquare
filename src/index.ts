require('isomorphic-fetch')

export * from './standalone'
export * from './utils'

export * from './DEPRECATED_actions'
export * from './DEPRECATED_reducers'
export * from './DEPRECATED_selectors'
