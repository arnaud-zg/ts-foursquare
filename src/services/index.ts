export * from './lists'
export * from './photos'
export * from './venues'
