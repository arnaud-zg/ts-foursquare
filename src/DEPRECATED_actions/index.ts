import { TLifeAction } from './life'

export type TRootAction = TLifeAction

export * from './life'
