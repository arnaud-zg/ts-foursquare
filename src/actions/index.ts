import { TLifeAction } from './life'
import { TListsAction } from './lists'
import { TPhotosAction } from './photos'
import { TVenuesAction } from './venues'

export type TRootAction =
  | TLifeAction
  | TListsAction
  | TPhotosAction
  | TVenuesAction

export * from './life'
export * from './lists'
export * from './photos'
export * from './venues'
