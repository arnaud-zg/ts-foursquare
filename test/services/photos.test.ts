import { getPhotosDetailsAsync } from '../../src/DEPRECATED_actions'
import { getObservablePhotosDetails } from '../../src/services'

describe('services/getObservablePhotosDetails', () => {
  it('should get an observable instance', () => {
    const action = getPhotosDetailsAsync.request({
      photoId: '51e4151c498e60b5d17bc721',
    })

    expect(getObservablePhotosDetails({ action })).toMatchSnapshot()
  })
})
