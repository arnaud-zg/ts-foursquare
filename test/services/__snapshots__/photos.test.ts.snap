// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`services/getObservablePhotosDetails should get an observable instance 1`] = `
Observable {
  "_isScalar": false,
  "operator": CatchOperator {
    "caught": [Circular],
    "selector": [Function],
  },
  "source": Observable {
    "_isScalar": false,
    "operator": SwitchMapOperator {
      "project": [Function],
    },
    "source": Observable {
      "_isScalar": false,
      "_subscribe": [Function],
    },
  },
}
`;
